src/nominatim_api/lookup.py

```diff
@@ -163,7 +164,7 @@ async def get_detailed_place(conn: SearchConnection, place: ntyp.PlaceRef,
 
     if collector.result is not None:
         await nres.add_result_details(conn, [collector.result], details)
-
+        await Formatter().localize_results([collector.result], details.locales)
```

Comment:
I think we should pull this even further out. Instead of doing it here, rather call it in server_glue.py after https://github.com/osm-search/Nominatim/blob/866e6bade9fad3a944bfcd05f450263c81b1ce49/src/nominatim_api/v1/server_glue.py#L246-L248

Then you can remove the initialisation of the Locale here at https://github.com/osm-search/Nominatim/blob/866e6bade9fad3a944bfcd05f450263c81b1ce49/src/nominatim_api/v1/server_glue.py#L230

That way, the Formatter can be initialised with the accept-langauges directly, having it effectively take over work of Locale.

    Formatter(get_accepted_languages(params)).localize_Results([collector.result])

That way, you can later write a replacement Formatter class that can do transliteration.